test/test_dataloader.py

```diff
@@ -2605,6 +2605,41 @@ def test_ind_worker_queue(self):
                 self._run_ind_worker_queue_test(batch_size=batch_size, num_workers=num_workers + 1)
 
 
+class TestMultiEpochDataset(IterableDataset):
+    def __init__(self, length):
+        self.length = length
+
+    def __iter__(self):
+        worker_info = torch.utils.data.get_worker_info()
+        assert worker_info is not None
+        worker_id = worker_info.id
+        assert self.length % worker_info.num_workers == 0
```

Comment:
What if the `length` cannot be divided by `num_workers` without remainder, does this test still work?

Reply:
it does work.  we may even be able to do this without batching.  i was being careful to avoid mixing a batch with data from different workers.

Reply:
Let me pull your PR later today to do an extra test.

Reply:
i pushed some updates